app/javascript/src/LoginPage/utils/formValidation.js

```diff
@@ -0,0 +1,64 @@
+import validate from 'validate.js'
+
+const constraintsTypes = {
+  text: {
+    presence: true,
+    length: {minimum: 1}
+  },
+  email: {
+    presence: true,
+    email: true,
+    length: {minimum: 1}
+  },
+  password: {
+    presence: true,
+    length: {minimum: 6}
+  },
+  password_confirmation: {
+    presence: true,
+    length: {minimum: 1},
+    equality: {
+      attribute: 'user[password]'
+    }
+  }
+}
+
+const constraints = {
+  'new_session': {
+    'username': constraintsTypes.text,
+    'password': constraintsTypes.password
+  },
+  'request_password': {
+    'email': constraintsTypes.email
+  },
+  'signup_form': {
+    'user[username]': constraintsTypes.text,
+    'user[email]': constraintsTypes.email,
+    'user[password]': constraintsTypes.password,
+    'user[password_confirmation]': constraintsTypes.password_confirmation
+  }
+}
+
+const validateAllFields = (form) => {
+  const errors = validate(form, constraints[form.id])
+
+  if (!errors) {
+    return undefined
+  } else {
```

Comment:
You've defined a type for Errors, use it treat it accordingly... like an empty Array of Error is cool, if not, show them.